pkg/collectors/devices/device_info.go

```diff
@@ -67,7 +67,7 @@ func extractOffsetFromTimestamp(result map[string]string) (map[string]any, error
 
 // BuildPTPDeviceInfo popluates the fetcher required for
 // collecting the PTPDeviceInfo
-func BuildPTPDeviceInfo(interfaceName string) error {
+func BuildPTPDeviceInfo(interfaceName string) error { //nolint:dupl //no need to dedup these
```

Comment:
I disagree, I think they can and should be deduped now by adding a generic fetcherFactory or fetcherBuilder.

I think it's two main changes:
* the params for each `AddNewCommand` get put into a slice so adding several commands to a fetcher is a case of iterating over that slice.
* `BuildPTPDeviceInfo` and `BuildDPLLInfoFetcher` call the builder that takes the above slice, a couple of other required bits (postProcessor, the command) as params, and returns the populated fetcherInst.

Reply:
I wanted more collectors before I de-duped them so I don't build a factory when need to throw it away for the next collector. But I'll give it a go.